### DISCUSSION TITLE
Feature Requests: Line return replacement and Option for a Text Down (Shrink to fit only) mode.

### DISCUSSION THREAD

**Discussion by celilo** [ChurchApps/ChurchAppsSupport#325] [FreeShow] [answered]
The following are two features that are really needed to replace PP in our use case. We use both of these extensively. Thanks in advance for your consideration.

Feature One:
Lower thirds present a challenge because text needs to be formatted in multiple ways to best utilize the available space. Below are a projected and lower third for a song in FreeShow.
<img width="1913" alt="FullGraphic" src="https://github.com/ChurchApps/FreeShow/assets/1428741/9c2e7991-ef5f-4a46-94fe-135c41330d0b">

<img width="1909" alt="Lower thirds too many lines" src="https://github.com/ChurchApps/FreeShow/assets/1428741/74477d23-cb61-49b1-959a-748098cd70d7">
This is not an extreme example, but the lower third presentation is poor. Because of the line returns, the text is placed on three lines and shrinks to fit. There is a great deal of wasted space in the text box. With more line returns, a small amount of text will result in additional rows with a very small font size.

ProPresenter handles this with a line return replacement feature. You can replace with spaces or select a character with surrounding spaces as a replacement. The result is a single line of text that wraps based on the size of the text box. This approach utilizes the lower thirds space far more efficiently. An example that replaces the line return with a diamond character: 
![FullvsLower3rds](https://github.com/ChurchApps/FreeShow/assets/1428741/c5664a77-ced5-4032-bd78-1935822098c2)

This looks way better and prevents the text from shrinking to an unreadable size.

Feature two:
Similarly,  a resize Text Down feature is needed to keep lower thirds from looking ridiculous. I set the text to my maximum desired size and apply a shrink to fit option in PP. The text autosize option in FreeShow has no upper limit, leading to very large text  when short and long phrases of text are used. This is the text size that I would like:
<img width="1902" alt="DesiredTextSize" src="https://github.com/ChurchApps/FreeShow/assets/1428741/f3c4565a-6184-4b8f-87bc-e6823beeb024">
This is what I get when there is a small amount of text.
<img width="1894" alt="bigText" src="https://github.com/ChurchApps/FreeShow/assets/1428741/40387b7b-8d55-465d-a050-72f43267732a">
It's quite disproportionate. With a shrink down only option, you output the exact desired font size, unless the text needs to shrink to fit.

**Comment by vassbo** [comment, marked as answer]
Thanks. Line break replacement would be a good feature to have.
You can already set the "Max auto font size" in the "Other" settings. :) (Location might change)

**Comment by celilo** [comment]
Thank you for pointing out the location of the "Max auto font size" setting. That is likely good enough to get to the next step for me, but it would be nice to have that as a textbox setting, since it is really dependent on what is being displayed.

**Comment by vassbo** [comment]
You will be able to set this per output style for the next version, so you can create different styles with different sized based on the content/scenario. Please note that the previous set value will be removed.

Edit: It will actually be replaced completely by this soon: https://github.com/ChurchApps/FreeShow/issues/761
